frontend/app/components/work-packages/wp-attachments/wp-attachments-edit.directive.html

```diff
@@ -2,38 +2,38 @@
   <div class="attributes-group--header">
     <div class="attributes-group--header-container">
       <h3 class="attributes-group--header-text">
-        {{ ::I18n.t('js.label_attachments') }}
+        {{ ::vm.I18n.t('js.label_attachments') }}
       </h3>
     </div>
   </div>
   <div class="work-package--attachments--files"
-       data-ng-show="attachments.length > 0">
+       data-ng-show="vm.attachments.length > 0">
     <div class="work-package--details--long-field">
       <span class="inplace-edit--read"
-            data-ng-repeat="attachment in attachments">
+            data-ng-repeat="attachment in vm.attachments track by attachment.id">
         <span class="inplace-editing--trigger-container">
           <span class="inplace-editing--trigger-link"
-                ng-class="{'-focus': focussing(attachment)}">
+                ng-class="{'-focus': vm.focussing(attachment)}">
             <span class="inplace-editing--container">
               <span class="inplace-edit--read-value">
                 <i class="icon-attachment"></i>
                 <a class="work-package--attachments--filename"
-                   data-ng-href="{{attachment._links.downloadLocation.href}}"
+                   data-ng-href="{{attachment._links.downloadLocation.href || '#'}}"
                    download
-                   data-ng-focus="focus(attachment)"
-                   data-ng-blur="focus(null)">
-                  {{::attachment.fileName}}
+                   data-ng-focus="vm.focus(attachment)"
+                   data-ng-blur="vm.focus(null)">
+                  {{::attachment.fileName || attachment.name}}
                 </a>
               </span>
               <a href=''
                  class="inplace-edit--icon-wrapper"
-                 data-ng-focus="focus(attachment)"
-                 data-ng-blur="focus(null)"
-                 data-ng-click="remove(attachment)"
-                 data-confirm-popup="{{I18n.t('js.text_attachment_destroy_confirmation')}}">
+                 data-ng-focus="vm.focus(attachment)"
+                 data-ng-blur="vm.focus(null)"
+                 data-ng-click="vm.remove(attachment)"
+                 data-confirm-popup="{{vm.I18n.t('js.text_attachment_destroy_confirmation')}}">
```

Comment:
It's been that way before, so not directed at you. This could become a one-time binding. You could also move this out to `scope.text`.